File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/StatusList.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import StatusItem from "@mateu/shared/apiClients/dtos/componentmetadata/StatusItem";

export default interface StatusList extends ComponentMetadata {
    items?: StatusItem[]
    compact?: boolean
    /** divider lines between rows but no outer border (the host provides the framing) */
    frameless?: boolean
    /** makes every row clickable: clicking one dispatches this action with { _item: id } */
    rowActionId?: string
    /** N-column responsive grid instead of a single stack; 0/absent = classic one-column list */
    columns?: number
    /** heading level of item titles in stacked mode (default 3 → h3; 4 → h4 under an h3 group) */
    itemHeadingLevel?: number
}

File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/StatusList.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

/**
 * A bordered list of rows with an icon, a title plus description, and a status chip and/or a small
 * action button on the right (see {@link StatusItem}) — incidents, side-effects checklists,
 * pending-tasks lists. An item's action dispatches its {@code actionId} with {@code { "_item": id
 * }}. {@code compact} tightens the row padding for dense screens; {@code frameless} keeps the
 * divider lines between rows but drops the outer border — for lists already framed by their host
 * (e.g. inside a section card). {@code rowActionId} makes every ROW clickable (search results, pick
 * lists): clicking one dispatches that action with {@code { "_item": id }}. Design-system neutral,
 * dark-mode aware.
 */
@Builder
public record StatusList(
    String id,
    List<StatusItem> items,
    boolean compact,
    boolean frameless,
    String rowActionId,
    String style,
    String cssClasses)
    implements Component {}
